src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/PlacementApplicationServiceTest.kt

```diff
@@ -585,6 +586,36 @@ class PlacementApplicationServiceTest {
       assertThat(entity.withdrawalReason).isEqualTo(PlacementApplicationWithdrawalReason.RELATED_APPLICATION_WITHDRAWN)
     }
 
+    @ParameterizedTest
+    @EnumSource(value = WithdrawableEntityType::class, names = ["Application","PlacementApplication"], mode = EnumSource.Mode.EXCLUDE)
```

Comment:
Yes your understanding is correct.

EnumSource.Mode.EXCLUDE means test all enum entries apart from the ones we list. i.e. the ones we list won't cause exceptions. 

It's a little clunky but i like it because it automatically tests for new enum types and at least alerts the developer they need to do something if behaviour is unexpected.

I'm going to rename the test because it's not really clear what we're testing. new name will be

'if withdraw is triggered by an entity that shouldn't cascade to placement applications, throws exception'

i'll do the same for other unit tests